nb2/models.py

```diff
@@ -40,6 +42,19 @@ def deserialize(self, data):
             if field in data:
                 setattr(self, field, data[field])
 
+    def has_said(self, quote:str) -> bool:
+        """
+        Check if quote already exists in Nostalgiabot's memory for this Person.
+
+        Args:
+            quote: a string representing something this Person said.
+
+        Returns:
+            True if a Quote record in the db for this Person has the same content
+            as quote. False otherwise
+        """
+        return any(q for q in self.quotes if q.content.lower()==quote.lower())
```

Comment:
Can leave this as is now, but I wonder if it's faster to do this through a db query

Reply:
Interesting. I would've thought this would be faster since it's presumably referencing a Person object already in memory? Not sure if that's actually what's going on

Reply:
Yeah, I think by default sqalchemy does load everything in one go. Sometimes memory stuff can be slower than db stuff though and I'm not sure if that would be the case here. We could do some fancier stuff with the quotes relationship to make it lazy or do some fancy dynamic filtering https://flask-sqlalchemy.palletsprojects.com/en/2.x/models/?highlight=lazy#one-to-many-relationships. It might be worth coming back to after we've built out more functionality to see if things work faster in different setups.

Reply:
Sounds good. And yeah at the moment, we don't really have a baseline to measure performance against